lib/Transforms/MemoryBanking.cpp

```diff
@@ -355,8 +475,8 @@ struct BankAffineLoadPattern
   }
 
 private:
-  uint64_t bankingFactor;
-  std::optional<int> bankingDimensionOpt;
+  mutable unsigned bankingFactor;
```

Comment:
Why is it necessary that these are mutable?

Reply:
Because `matchAndOverwrite` has `const` in its signature: https://github.com/llvm/circt/blob/335e50ce67b802b20ea960578e37f4950df233e5/lib/Transforms/MemoryBanking.cpp#L295